R/data_color.R

```diff
@@ -413,18 +428,71 @@ get_alpha_vec <- function(colors) {
 
   for (color in colors) {
 
-    if (grepl("^#[0-9A-Fa-f]{6}$", color)) {
+    if (color %>% is_hex_col_no_alpha()) {
       alpha <- c(alpha, "FF")
-    } else if (grepl("^#[0-9A-Fa-f]{8}$", color)) {
+    } else if (color %>% is_hex_col_w_alpha()) {
       alpha <- c(alpha, toupper(sub(".*(..)$", "\\1", color)))
-    } else {
+    } else if (tolower(color) %in% grDevices::colors()) {
       alpha <- c(alpha, "FF")
+    } else {
+      alpha <- c(alpha, NA_character_)
     }
   }
 
   alpha
 }
 
+#' Get the hexadecimal color representation of a color
+#'
+#' @noRd
+get_hexcolor_vec <- function(colors) {
+
+  hexcolors <- c()
+
+  for (color in colors) {
+
+    if (color %>% is_hex_col_no_alpha()) {
+      hexcolors <- c(hexcolors, color %>% toupper())
+    } else if (color %>% is_hex_col_w_alpha()) {
+      hexcolors <- c(hexcolors, color %>% substring(1, 7) %>% toupper())
+    } else if (tolower(color) %in% grDevices::colors()) {
+
+      color_matrix <- grDevices::col2rgb(tolower(color))
+      color <- grDevices::rgb(
+        red = color_matrix[1, ]/255,
+        green = color_matrix[2, ]/255,
+        blue = color_matrix[3, ]/255
+      )
+
+      hexcolors <- c(hexcolors, color)
+    } else {
+      hexcolors <- c(hexcolors, NA_character_)
+    }
+  }
+
+  hexcolors
+}
+
+#' With any input color, transform to hexadecimal with alpha
+#'
+#' @noRd
+normalize_color <- function(color,
+                            alpha) {
+
+  # Get hexadecimal representation of `color`; this strips
+  # any alpha value that may be included
+  color_hex <- color %>% get_hexcolor_vec()
```

Comment:
This should be something like...

```r
  if (!is_hex_col_w_alpha(color)) {
    alpha_hex <- alpha %>% frac_alpha_to_hex()
    color <- paste0(get_hexcolor_vec(color), frac_alpha_to_hex(alpha))
  }

  color
```

This should only update color if necessary (my thought)

Reply:
this should work with a vector of color vals